s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/SourceRecordIterator.java

```diff
@@ -136,24 +134,34 @@ private List<S3SourceRecord> readNext() {
 
                 int numOfProcessedRecs = 1;
                 boolean checkOffsetMap = true;
-                for (final Object record : transformer.getRecords(valueInputStream, topic, topicPartition,
-                        s3SourceConfig)) {
-                    if (offsetManager.shouldSkipRecord(partitionMap, currentObjectKey, numOfProcessedRecs)
-                            && checkOffsetMap) {
-                        numOfProcessedRecs++;
-                        continue;
-                    }
+                try (Stream<Object> recordStream = transformer.getRecords(s3Object::getObjectContent, topic,
+                        topicPartition, s3SourceConfig)) {
+                    final Iterator<Object> recordIterator = recordStream.iterator();
+                    while (recordIterator.hasNext()) {
+                        final Object record = recordIterator.next();
+
+                        // Check if the record should be skipped based on the offset
+                        if (offsetManager.shouldSkipRecord(partitionMap, currentObjectKey, numOfProcessedRecs)
+                                && checkOffsetMap) {
+                            numOfProcessedRecs++;
+                            continue;
+                        }
+
+                        final byte[] valueBytes = transformer.getValueBytes(record, topic, s3SourceConfig);
+                        checkOffsetMap = false;
+
+                        sourceRecords.add(getSourceRecord(keyBytes, valueBytes, topic, topicPartition, offsetManager,
```

Comment:
If we add the source record here do we need to increment numOfProcessedRecs just below before we check if we have reached the max poll records?

Reply:
Actually it would not affect if the loop breaks. But for readability, moved the increment to before max poll recs check

Reply:
cool, thank you